client/state/reader/site-blocks/actions.js

```diff
@@ -0,0 +1,80 @@
+/**
+ * External dependencies
+ */
+
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+import {
+	READER_SITE_BLOCK_REQUEST,
+	READER_SITE_BLOCK_REQUEST_SUCCESS,
+	READER_SITE_BLOCK_REQUEST_FAILURE,
+	READER_SITE_UNBLOCK_REQUEST,
+	READER_SITE_UNBLOCK_REQUEST_SUCCESS,
+	READER_SITE_UNBLOCK_REQUEST_FAILURE,
+} from 'state/action-types';
+
+/**
+ * Triggers a network request to block a site.
+ *
+ * @param  {Integer} siteId Site ID
+ * @return {Function} Action thunk
+ */
+export function requestSiteBlock( siteId ) {
```

Comment:
I agree that'd it'd be nice to DRY that up @samouri! Sounds like something we should tackle on a framework level. Fancy whipping up a PR?